xkcd.py

```diff
@@ -0,0 +1,36 @@
+import re
+import requests
+from bs4 import BeautifulSoup
+import random
+import os
+
+
+l=[] #For storing random numbers
+
+
+def download_img(data,filename): #Function to download images
+    if (os.path.isdir('XKCD')): #Asserts for existence
+        pass
+    else:
+        os.mkdir('XKCD') #If false create a folder
+    op_file=open('XKCD/'+filename,'wb')
+    op_file.write(data) #Download off
+    print "Downloaded",filename
+
+
+choice=input("How many random images you want to download?? xD \n")
+
+for i in range(choice):
```

Comment:
there is no need of this loop and no need of `l=[]`
you could simply generate a random number on the go.